crates/language/src/language_registry.rs

```diff
@@ -710,18 +716,9 @@ impl LanguageRegistry {
         self: &Arc<Self>,
         path: &Path,
         content: Option<&Rope>,
-        user_file_types: Option<&FxHashMap<Arc<str>, GlobSet>>,
+        user_file_types: Option<&FxHashMap<Arc<str>, LanguageCustomFileTypes>>,
     ) -> Option<AvailableLanguage> {
         let filename = path.file_name().and_then(|name| name.to_str());
-        // `Path.extension()` returns None for files with a leading '.'
-        // and no other extension which is not the desired behavior here,
-        // as we want `.zshrc` to result in extension being `Some("zshrc")`
-        let extension = filename.and_then(|filename| filename.split('.').next_back());
-        let path_suffixes = [extension, filename, path.to_str()];
-        let path_suffixes_candidates = path_suffixes
-            .iter()
-            .filter_map(|suffix| suffix.map(globset::Candidate::new))
-            .collect::<SmallVec<[_; 3]>>();
```

Comment:
I think we should restore this. The purpose was to avoid repeatedly constructing match candidates (which is done by `GlobSet::matches`).

Reply:
Done. Note that I changed this so that `path_suffixes` is now a small vec of `(suffix, candidate)` tuples. This makes is easy to keep matching on the candidate, while also making it easy to reference the suffix for that candidate.